local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -100,15 +100,36 @@ func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConf
 			continue
 		}
 		filePath := filepath.Join(config.MountDir, file)
-		err = d.validateFile(filePath)
-		if err != nil {
-			glog.Errorf("Mount path %q validation failed: %v", filePath, err)
-			continue
-		}
-		availByte, err := d.VolUtil.GetFsAvailableByte(filePath)
-		if err != nil {
-			glog.Errorf("Path %q fs stats error: %v", filePath, err)
-			continue
+		var availByte uint64
+		if config.VolumeType == common.VolumeTypeBlock {
+			var ret bool
+			ret, err = d.VolUtil.IsBlock(filePath)
```

Comment:
@msau42 How about I follow the same pattern as the validateFile() method and create a validateBlock()  method here? It will return just an err object as outlined by you. I would like to keep the IsBlock() function with its lower level interface that distinguishes a request failure from boolean return value as it provides more flexibility to the caller.